src/components/Product-card/ProductCard.jsx

```diff
@@ -23,6 +29,18 @@ const ProductCard = () => {
                   dispatchOfProductsBeingShown({ type: "filterByCategory", payload: category })
                 }
         }
+
+        if( newArrivalCategory ){
+            dispatchOfProductsBeingShown({ type: "filterByProducts", payload: "isNewArrival" })
+        }
+
+        if( discountedProducts ){
+            dispatchOfProductsBeingShown({ type: "filterByProducts", payload: "hasDiscount" })
+        }
+
+        if( productOnSale ){
```

Comment:
Multiple If's hm... Have you tried ternary operator or the logical and operator?

Reply:
yes i thought it would increase the readability as i was using nesting conditions here!!!